From e19d36f215191f7a313380a3f25418a0378ab741 Mon Sep 17 00:00:00 2001
From: Kenny Kessler <kenny.kessler@motorola.com>
Date: Thu, 28 May 2015 11:10:42 -0500
Subject: [PATCH 262/959] IKSWL-19229: power: smbcharger-mmi: Fix ESR Pulse
 Reset

The current is not returning to the correct Parallel
setting after the ESR pulse is generated.  Fix this
by checking the Parallel Current setting is not zero
and then running the target current through the split
table.

Change-Id: I524a01841ee22cca6f87aeb6015f6436b216d9b1
Signed-off-by: Kenny Kessler <kenny.kessler@motorola.com>
Reviewed-on: http://gerrit.mot.com/752239
SME-Granted: SME Approvals Granted
SLTApproved: Slta Waiver <sltawvr@motorola.com>
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Ravikumar Vembu <raviv@motorola.com>
Reviewed-by: Vidyadhar Gurram <rnh786@motorola.com>
Reviewed-by: Jason Knopsnyder <jasonk@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 drivers/power/qpnp-smbcharger-mmi.c | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/drivers/power/qpnp-smbcharger-mmi.c b/drivers/power/qpnp-smbcharger-mmi.c
index de831d9b90b..8c91805b2a2 100644
--- a/drivers/power/qpnp-smbcharger-mmi.c
+++ b/drivers/power/qpnp-smbcharger-mmi.c
@@ -1824,12 +1824,20 @@ static int smbchg_parallel_usb_charging_en(struct smbchg_chip *chip, bool en)
 static int smbchg_sw_esr_pulse_en(struct smbchg_chip *chip, bool en)
 {
 	int rc;
+	int index = 0;
+	int target_current = chip->target_fastchg_current_ma;
+
+	if (chip->parallel.current_max_ma != 0) {
+		index = smbchg_get_pchg_current_map_index(chip);
+		target_current = chip->pchg_current_map_data[index].primary;
+	}
 
 	chip->sw_esr_pulse_en = en;
-	rc = smbchg_set_fastchg_current(chip, chip->target_fastchg_current_ma);
+	rc = smbchg_set_fastchg_current(chip, target_current);
 	if (rc)
 		return rc;
-	rc = smbchg_parallel_usb_charging_en(chip, !en);
+	if (chip->parallel.current_max_ma != 0)
+		rc = smbchg_parallel_usb_charging_en(chip, !en);
 	return rc;
 }
 
-- 
2.11.0

